lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Popup/PopupModal.jsx

```diff
@@ -4,16 +4,26 @@ import * as classNames from 'classnames';
 import {Modal, Button, Hyperlink} from '@edx/paragon/static';
 
 export const PopupModal = ({open, onHide, message, error, inProgress, succeeded, reportPath}) => {
-    const handleDownload = () => {
-        window.open(`/media/${reportPath}`, '_blank');
-    };
 
-    const buttons = reportPath !== null
-        ? [<Button label={gettext("Download CSV")} onClick={handleDownload}/>]
+    const previewButton =(
+        <Button label={gettext("Preview CSV")}
+                onClick={() =>
+                    window.open(`/csv-viewer?csvUrl=${reportPath}`, '_blank')
+                }/>
+    );
+    const downloadButton =(
+        <Button label={gettext("Download CSV")}
+                onClick={() =>
+                    window.open(reportPath, '_blank')
+                }/>
+    );
+
+    const buttons = (reportPath === null && !error)
```

Comment:
Shouldn't this be `(reportPath !== null && !error)` ? i.e. Doesn't `reportPath` get set once the report is available?

Reply:
Yup, sorry. I think I made an amended commit for this but forgot to push it.